base/sparse/sparsematrix.jl

```diff
@@ -1236,6 +1234,12 @@ end
 getindex{T<:Integer}(A::SparseMatrixCSC, I::AbstractVector{T}, j::Integer) = getindex(A,I,[j])
 getindex{T<:Integer}(A::SparseMatrixCSC, i::Integer, J::AbstractVector{T}) = getindex(A,[i],J)
 
+# Colon translation (this could be done more efficiently)
```

Comment:
The real problem is anyone _using_ these methods. But yes, someone (not you) should eventually build faster implementations.